src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -191,3 +204,54 @@ func (q *qemuAmd64) appendImage(ctx context.Context, devices []govmmQemu.Device,
 func (q *qemuAmd64) appendBridges(devices []govmmQemu.Device) []govmmQemu.Device {
 	return genericAppendBridges(devices, q.Bridges, q.qemuMachine.Type)
 }
+
+// enable protection
+func (q *qemuAmd64) enableProtection() error {
+	flags, err := CPUFlags(procCPUInfo)
+	if err != nil {
+		return err
+	}
+
+	// TDX is supported and properly loaded when the firmware directory exists or `tdx` is part of the CPU flags
+	if d, err := os.Stat(tdxSysFirmwareDir); (err != nil && d.IsDir()) || flags[tdxCPUFlag] {
+		q.protection = tdxProtection
+		if q.qemuMachine.Options != "" {
+			q.qemuMachine.Options += ","
+		}
+		q.qemuMachine.Options += "kvm-type=tdx,confidential-guest-support=tdx"
+		q.kernelParams = append(q.kernelParams, Param{"tdx_guest", ""})
+		virtLog.WithFields(logrus.Fields{
+			"subsystem":     "qemuAmd64",
+			"machine":       q.qemuMachine,
+			"kernel-params": q.kernelParameters}).
+			Info("Enabling TDX guest protection")
+		return nil
+	}
+
+	// TODO: Add support for other x86_64 technologies: SEV
+
+	return fmt.Errorf("This system doesn't support Confidential Computing (Guest Protection)")
+}
+
+// add protection device
+func (q *qemuAmd64) addProtectionDevice(devices []govmmQemu.Device, firmware string) ([]govmmQemu.Device, bool, error) {
```

Comment:
To be consistent with the rest of the `qemuArch` interface, I'd call that one `appendProtectionDevice`.

Also, I would make it use the following prototype:

```Go
func (q *qemuAmd64) appendProtectionDevice(devices []govmmQemu.Device, bios string) ([]govmmQemu.Device, string, error) {
```

So that it returns what the bios path should be:

```Go
qemuConfig.Devices, qemuConfig.Bios, err = q.arch.appendProtectionDevice(qemuConfig.Devices, qemuConfig.Bios)
```

The logic of using or not a bios would be embedded into the arch specific function, and it simplifies the `if q.config.GuestProtection {` block by not having to explicitly clearing the bios path or not.